src/Css/Style.php

```diff
@@ -1640,6 +1688,12 @@ protected function compute_prop(string $prop, $val)
             return $val;
         }
 
+        // Resolve CSS custom property value.
+        if (substr($val, 0, 4) === 'var(') {
+            $var = substr(trim($val), 4, -1);
+            return $this->parse_var($var);
+        }
```

Comment:
This assumes the entirety of the value is a variable reference. But that may not be true, for example:

```
<style>
    :root {
        --bg: orange;
    }
    div {
        border: 2px solid var(--bg);
    }
</style>
<div>dompdf</div>
```

Reply:
True, or with 2 vars:  `border: 2px vra(--border-style) var(--bg);`

Reply:
Resolved by using preg_replace_callback().

Reply:
The cited change doesn't work on shorthand properties because the CSS variable isn't resolved prior to applying the value to the more specific CSS properties. In this particular example since `var(--bg)` isn't resolved it doesn't match a valid border color value and so the entire declaration is dropped.

First thing to do would be to add variable resolution for shorthand properties. If we do that in `set_prop` before calling the shorthand method then the variable reference parses.

The next thing to do is more of a challenge. If the variable value changes after the shorthand has been parsed we need to call `set_prop` on that shorthand again to propagate the value down to the specific CSS properties. However, if we did a straight up variable dependency, then something like the following:

```
<style>
    div {
        --color: orange;
        border: 2px solid var(--color);
        border-color: blue;
    }
</style>
```

will get the wrong result because re-processing the `border` property will overwrite what was set by `border-color`.

Perhaps the easiest solution to this scenario would be to completely re-process the standard properties that have been set whenever a custom property value is set. It's a bit inefficient but I did some quick testing and it works.